articlequality/extractors/ptwiki.py

```diff
@@ -6,24 +6,58 @@
 logger = logging.getLogger(__name__)
 
 
+POSSIBLE_LABELS = {
+    "1": "1",
+    "2": "2",
+    "3": "3",
+    "4": "4",
+    "5": "5",
+    "6": "6",
+    "ab": "5",
+    "ad": "6"
+}
+
+
 def from_template(template):
-    project_name = normalize_project_name(template.name)
-    if project_name == "marca de projeto":
-        labels = extract_labels(template)
-        if len(labels) >= 1:
-            for project, label in labels[1:]:
-                yield (normalize_project_name(project), label)
+    template_name = normalize_template_name(template.name)
+    if template_name == "marca de projeto":
+        label = extract_label(template)
+        if label is not None:
+            yield ("marca de projeto", label)
+
 
+def extract_label(template):
+    # Try to get the label from {{{1}}}
+    try:
+        label = normalize_label(template.get(1).value)
+    except ValueError:
+        label = None
+    if label is None or label == "?":
+        # Try to get the label from {{{qualidade}}}
+        try:
+            label = normalize_label(template.get("qualidade").value)
+        except ValueError:
+            label = label or None
+    # If we failed, log a warning
+    if label is None:
+        logger.warn("Could not extract label from {0}".format(str(template)))
+        return None
+    else:
+        return label
 
```

Comment:
I was thinking in something along the lines of this (untested):
```python
def extract_label(template):
    # Try to get the label from {{{1}}}
    try:
        label = template.get(1).value or template.get("qualidade").value
    except ValueError:
        label = None
    n_label = normalize_label(label)
    # If we failed, but it is not "?"'s fault, log a warning
    if n_label is None and label != "?":
        logger.warn("Could not extract label from {0}".format(str(template)))
    return n_label

def normalize_label(label):
    label = label.lower()
    if label in POSSIBLE_LABELS:
        return POSSIBLE_LABELS[label]
    else:
        return None
```

Reply:
We don't want to return "?" as a valid label.  It's as valid of a label as "" (empty string) or "foobar".

Reply:
The return value `n_label` is never `?` in the example above, because it is the normalization of `label` (if `label` was `?`, `n_label` would be `None`)

Reply:
Aha!  I see you're checking `label` in that conditional.

Reply:
On a side note, I wish that something logical like  `template.get(1).value or template.get("qualidade").value` would work but it results in a ValueError when {{{1}}} is empty but "{{{qualidade}}}" is set.   Otherwise, I think our logic matches here.